src/bin/cargo/commands/remove.rs

```diff
@@ -0,0 +1,122 @@
+        .args([clap::Arg::new("dependencies")
+            .action(clap::ArgAction::Append)
+            .required(true)
+            .multiple_values(true)
+            .takes_value(true)
+            .value_name("DEP_ID")
+            .help("Dependencies to be removed")])
+        .arg_package("Package to remove from")
+        .arg_manifest_path()
+        .arg_quiet()
+        .arg_dry_run("Don't actually write the manifest")
+        .next_help_heading("SECTION")
+        .args([
+            clap::Arg::new("dev")
+                .long("dev")
+                .conflicts_with("build")
+                .action(clap::ArgAction::SetTrue)
+                .group("section")
+                .help("Remove as development dependency"),
+            clap::Arg::new("build")
+                .long("build")
+                .conflicts_with("dev")
+                .action(clap::ArgAction::SetTrue)
+                .group("section")
+                .help("Remove as build dependency"),
+            clap::Arg::new("target")
+                .long("target")
+                .takes_value(true)
+                .value_name("TARGET")
+                .value_parser(clap::builder::NonEmptyStringValueParser::new())
+                .help("Remove as dependency from the given target platform"),
+        ])
+}
+
+pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
+    let dry_run = args.dry_run();
+
+    let workspace = args.workspace(config)?;
+    let packages = args.packages_from_flags()?;
+    let packages = packages.get_packages(&workspace)?;
+    let spec = match packages.len() {
+        0 => {
+            return Err(CliError::new(
+                anyhow::format_err!("no packages selected. Please specify one with `-p <PKG_ID>`"),
+                101,
+            ));
+        }
+        1 => packages[0],
+        len => {
+            return Err(CliError::new(
+                anyhow::format_err!(
+                    "{len} packages selected. Please specify one with `-p <PKG_ID>`",
+                ),
+                101,
+            ));
+        }
+    };
+
+    let dependencies = args
+        .get_many::<String>("dependencies")
+        .expect("required(true)")
+        .cloned()
+        .collect();
+
+    let section = parse_section(args);
+
+    let options = RemoveOptions {
+        config,
+        spec,
+        dependencies,
+        section,
+        dry_run,
+    };
+    remove(&options)?;
+
+    if !dry_run {
+        // Reload the workspace since we've changed dependencies
+        let ws = args.workspace(config)?;
+        resolve_ws(&ws)?;
+    }
+
+    Ok(())
+}
+
+fn parse_section(args: &ArgMatches) -> DepTable {
+    let dev = args
+        .get_one::<bool>("dev")
+        .copied()
+        .expect("action(ArgAction::SetTrue)");
+    let build = args
+        .get_one::<bool>("build")
+        .copied()
+        .expect("action(ArgAction::SetTrue)");
```

Comment:
nit:

1. The prelude defines an `ArgMatchesExt::flag` for simplifying these calls
2. Cargo hasn't been updated to it but the latest clap has that has `get_flag`

Reply:
Now fixed to use `ArgMatchesExt::flag`.